### ISSUE TITLE
[iOS] `Resuming` event is triggered when application is launched

### ISSUE THREAD

**Issue by antoine-hebert** [unoplatform/uno#10313]
### Current behavior

After upgrading to 4.x from version 3.X we noticed a significant change in the application lifecycle in regards to the Resuming event.

On 4.x **Resuming** event is now being triggered during application startup, before **OnLaunched** event which is not expected.

In addition, although this should probably be filed as another issue, we also noticed that **Resuming** is now triggered twice when an application actually resumes.

### Expected behavior

We're not expecting Resuming event to be triggered when application is launched.

We are also expecting Resuming event to be triggered only once when the app resumes.

On a side note, it would be great if application lifecycle could be documented.

### How to reproduce it (as minimally and precisely as possible)

Launch the attached repro project, look at App.Xaml.cs. 

**1. Launch app**

> 2022-11-02 15:37:24.749 test_lifecycle[18667:8079307] OnResuming
> 2022-11-02 15:37:24.749 test_lifecycle[18667:8079307] OnLaunched

**2. Send app to background**
**3. Bring app to foreground**

> 2022-11-02 15:37:56.506 test_lifecycle[18667:8079307] OnResuming
> 2022-11-02 15:37:56.558 test_lifecycle[18667:8079307] OnResuming
> 2022-11-02 15:37:56.560 test_lifecycle[18667:8079307] WillEnterForeground

[uno_test_lifecycle.zip](https://github.com/unoplatform/uno/files/9923891/uno_test_lifecycle.zip)

### Workaround

We are able to workaround this issue by overriding WillEnterForeground on iOS.

### Environment

Uno.SourceGenerationTasks

### Affected platforms

Android, iOS

### IDE

Visual Studio 2022

### IDE version

17.3.6
